### PR TITLE
Dg/hci modified

### PR DESCRIPTION
**Pull request by MastroDiego**
This update introduces a programmable-length FIFO job queue to the source and sink modules. The queue allows jobs to be stored and processed sequentially, enabling autonomous job management. Each module can now execute jobs without direct control, improving flexibility in job handling.

### CODE REVIEW

**Comment by FrancescoConti** [comment]
Hi Diego, thanks for the PR! Generally looks very good, I left you a few comments, all of them minor.
How are you testing this, on RedMulE/SoftEx/other?

**Comment by MastroDiego** [comment]
Hi Francesco, thanks for your feedback! I have tested this on HWPE-Datamover only to validate edge cases. The plan is to implement it on Neureka.

**Comment by FrancescoConti** [comment]
OK, I suggest to push the tests to HWPE-Datamover and prepare a PR there, it could be useful.

**Comment by MastroDiego** [comment]
I made some quick and not so pretty changes in the Datamover (I increased the number of registers in the control slave to simulate multiple jobs offloading at the same time). That's why Andrea and I initially thought it wouldn't make sense to open a PR. But if needed, I can clean things up and make it more structured.

**Comment by FrancescoConti** [comment]
Maybe open a draft PR as is, just not to lose track ;)

File: Bender.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@ sources:
       - rtl/core/hci_core_mux_ooo.sv
       - rtl/core/hci_core_r_valid_filter.sv
       - rtl/core/hci_core_r_id_filter.sv
-      - rtl/core/hci_core_source.sv
```
Comment (belanoa) [inline Bender.yml:36]: Add this line back: it is not possible to use the old `hci_core_source` at the moment
